test/Unit/StringFormatTests.ts

```diff
@@ -0,0 +1,21 @@
+import {} from '../../src/Constants';
+import * as assert from 'assert';
+
+describe('String.format Tests', () => {
+    it('Bad template returns same string', () => {
+        const template = 'template {} {asdf}';
+        assert.equal(String.format(template, 1, 2, 3), template);
+    });
+
+    it('Replace patterns correctly', () => {
+        const template = 'template {0} {1} template';
+        assert.equal(String.format(template, 'a', 'b', 'c'), 'template a b template');
+        assert.equal(String.format(template, 'a', 'b'), 'template a b template');
+        assert.equal(String.format(template, 'a'), 'template a  template');
+    });
+    
+    it('Will work if param is same as pattern', () => {
```

Comment:
have a test case withut {}

Reply:
and one with only { and one with only }

---
In reply to: [256706917](https://github.com/karanjitsingh/jstestcontext/pull/7#discussion_r256706917) [](ancestors = 256706917)

Reply:
it seems you don't trust regex :P

Reply:
do i trust bcl, do i trust node do i trust ur code. if everything is on trust then we dont need tests :D

---
In reply to: [256707247](https://github.com/karanjitsingh/jstestcontext/pull/7#discussion_r256707247) [](ancestors = 256707247)

Reply:
updated